handlers/feed.go

```
@@ -61,84 +68,88 @@ func DownloadYoutubeFeed(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	var dataCount int64 = 0
 	for i := 0; i < len(youtube_download.YoutubeUrls); i++ {
 		url := youtube_download.YoutubeUrls[i]
+		// Split URL to get video ID
+		idSplit := strings.Split(url, "v=")
+		id := idSplit[1]
 
-		feed, err := feeds.ParseFeed(url, false)
-		if err == nil {
-			processYoutubeDownload(url, *feed)
-		}
+		// Make a Youtube API call to check if the video exists
+		call := tube.Videos.List([]string{"snippet,contentDetails,statistics"}).Id(id)
+		response, _ := call.Do()
+
+		// Add the Youtube results to the data result it should be one if the video exists
+		dataCount += response.PageInfo.TotalResults
 	}
 
-	w.WriteHeader(http.StatusOK)
-	json.NewEncoder(w).Encode("Youtube download processed successfully")
+	if int(dataCount) == len(youtube_download.YoutubeUrls) {
+		processYoutubeDownload(youtube_download.YoutubeUrls)
+		w.WriteHeader(http.StatusOK)
+		json.NewEncoder(w).Encode("Youtube download processed successfully")
+	} else {
+		w.WriteHeader(http.StatusBadRequest)
+		json.NewEncoder(w).Encode("Could not process Youtube download, one of the youtueb videos does not exists")
```

can we do this in the for loop? if one URL failing causes this to fail we should fail fast instead of going through all the URLs and then failing

---

we could then avoid the dataCount variable